src/PrestaShopBundle/Form/Admin/Sell/Product/Options/ProductAttachmentsType.php

```diff
@@ -0,0 +1,112 @@
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class ProductAttachmentsType extends TranslatorAwareType
+{
+    /**
+     * @var UrlGeneratorInterface
+     */
+    private $urlGenerator;
+
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        UrlGeneratorInterface $urlGenerator
+    ) {
+        parent::__construct($translator, $locales);
+        $this->urlGenerator = $urlGenerator;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('attached_files', EntitySearchInputType::class, [
+                'entry_type' => AttachedFileType::class,
+                'layout' => EntitySearchInputType::TABLE_LAYOUT,
+                'required' => false,
+                'label' => false,
+                'remote_url' => $this->urlGenerator->generate('admin_attachments_search', ['searchPhrase' => '__QUERY__']),
+                'placeholder' => $this->trans('Search file', 'Admin.Catalog.Feature'),
+                'empty_state' => $this->trans('No files associated', 'Admin.Catalog.Feature'),
+            ])
+            ->add('add_attachment_btn', IconButtonType::class, [
+                'label' => $this->trans('Add new file', 'Admin.Catalog.Feature'),
+                'icon' => 'add_circle',
+                'type' => 'link',
+                'attr' => [
+                    'data-success-create-message' => sprintf('%s %s',
+                        $this->trans(
+                            'Attachment was successfully created and added to the selection.',
+                            'Admin.Catalog.Feature'
+                        ),
+                        $this->trans(
+                            'This window will close automatically.',
+                            'Admin.Global'
+                        )
+                    ),
+                    'class' => 'btn-outline-secondary add-attachment',
+                    'href' => $this->urlGenerator->generate('admin_attachments_create', [
+                        'liteDisplaying' => true,
+                        'saveAndStay' => true,
+                    ]),
+                ],
+            ])
+        ;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        parent::configureOptions($resolver);
+        $resolver->setDefaults([
+            'label' => $this->trans('Attached files', 'Admin.Catalog.Feature'),
+            'label_tag_name' => 'h2',
+            'label_subtitle' => $this->trans('Select the files (instructions, documentation, recipes, etc.) your customers can directly download on this product page.', 'Admin.Catalog.Feature') .
```

Comment:
I think this could be improved (maybe separate it into a tooltip and a help text) but this needs to be improved by a designer, so we'll change it later if needed. :-)